### PR TITLE
Update DOI award box for new use cases

### PR DESCRIPTION
**Pull request by JamesTessmer**
This is to address issue https://github.com/microbiomedata/submission-schema/issues/224

This PR will add some changes to the Dataset DOI textbox such as changing the name, making it multi-valued, and allowing it to be used when JGI or EMSL are selected in the context form.

### CODE REVIEW

**Comment by JamesTessmer** [comment]
I think one of the requirements from https://github.com/microbiomedata/submission-schema/issues/224 is to make the text box multivalued.
I'm not sure if we want to approach that the same way we did for funding source or if we wanted to do it another way.

**Comment by JamesTessmer** [comment]
@pkalita-lbl @mslarae13 I think I've captured all the things we talked about in the meeting last week.
The new commits makes `awardDois` multivalued and there are some new functions enforcing the requirements.
This includes:
 - When a user checks the JGI/EMSl checkboxes a new textbox appears (2 will appear if they check both)
 - When a user checks the `data was generated by a DOE user facility` a new textbox appears
 - There are rules enforcing the number of `awardDois`
 -  - If JGI/EMSL are checked users cannot remove any `awardDois` that would reduce them to less than the number of facilities checked, so if both are checked they cannot go below 2, if one is checked they can't go below one. 
 - - If `data was generated by a DOE user facility` is checked they cannot go below a single `awardDois`

I think this should be ready for review. I expect there will be a couple of minor things but let me know if I missed any big things or need to reword something.

File: nmdc_server/schemas_submission.py
Comment (pkalita-lbl) [inline nmdc_server/schemas_submission.py:73]: Recommend consistent casing:
```suggestion
    unknownDoi: Optional[bool]
```

File: web/src/definitions.ts
Comment (pkalita-lbl) [inline web/src/definitions.ts:20]:
```suggestion
  unknownDoi: 'If selected you will not have to enter the required Award DOIs and you will be contacted by an NMDC team member after you have submitted your data.',
```

File: web/src/views/SubmissionPortal/store/index.ts
Comment (pkalita-lbl) [inline web/src/views/SubmissionPortal/store/index.ts:117]:
```suggestion
  unknownDoi: undefined as undefined | boolean,
```

**Review by pkalita-lbl — COMMENTED** [review]
Overall the changes look good. Just a few stylistic comments.

My main question is probably for @mslarae13: If a submission previously had a value entered into the field labeled Dataset DOI

image

Should that value be carried over into the updated filed now labeled Award DOI?

image

**Comment by JamesTessmer** [comment]
@pkalita-lbl I pushed up all the requested changes for  unknownDoi
@mslarae13 With that done I think you just need to double check what Patrick mentioned and then we should be done

**Comment by mslarae13** [comment]
> Overall the changes look good. Just a few stylistic comments.
> 
> My main question is probably for @mslarae13: If a submission previously had a value entered into the field labeled Dataset DOI
> 
> image
> Should that value be carried over into the updated filed now labeled Award DOI?
> 
> image

@pkalita-lbl @JamesTessmer ... yes. Keep the same information and just carry it over

**Comment by JamesTessmer** [comment]
@pkalita-lbl Would changing the variable from `awardDois` back to `datasetDoi` accomplish that?

**Comment by mslarae13** [comment]
Decision today at infrastructure sync: will not merge until after berkeley-schema is in production, so after oct 15. There will be merge conflicts

**Review by marySalvi — CHANGES_REQUESTED** [review]
The commit history could use some work.

**Comment by JamesTessmer** [comment]
@marySalvi I was planning to squash the recent commits, I was pushing them up for some bug fixing, once the issue I’m running into is resolved. Were there other commits you wanted cleaned up as well? I’m happy to meet and talk about that and if you have a preferred method you want used here.

**Comment by marySalvi** [comment]
> @marySalvi I was planning to squash the recent commits, I was pushing them up for some bug fixing, once the issue I’m running into is resolved. Were there other commits you wanted cleaned up as well? I’m happy to meet and talk about that and if you have a preferred method you want used here.

It might be a good idea to meet. Can I also suggest running tox locally?
`tox -e [test environment]`

ie `tox -e format,lint`

**Comment by JamesTessmer** [comment]
> > @marySalvi I was planning to squash the recent commits, I was pushing them up for some bug fixing, once the issue I’m running into is resolved. Were there other commits you wanted cleaned up as well? I’m happy to meet and talk about that and if you have a preferred method you want used here.
> 
> It might be a good idea to meet. Can I also suggest running tox locally? `tox -e [test environment]`
> 
> ie `tox -e format,lint`

I usually do run tox locally (sometimes I forget to apply the linting) but in this case typecheck was giving me a different error than what typecheck from github actions was giving. I probably should have resolved that first but I was hoping I could fix the issue with this code last night first and squash the commits.
I can slack you some meeting times so we can talk about it.

**Review by marySalvi — APPROVED** [review]

**Comment by JamesTessmer** [comment]
@pkalita-lbl @mslarae13 Now that the schema is updated is this ok to start finalizing the merge? I know we need to merge main back in and there might be some merge conflicts. I did noticed that there were two new migrators in main now and one revises the same version as the migrator added in this PR
`Revises: 5fb9910ca8e6`
Do we need to remake the migrator file so that it revises the most recent version and we don't have 2 migrators revising the same thing?

File: nmdc_server/migrations/versions/988105f6581d_update_datasetDoi_to_awardDoi.py
Comment (marySalvi) [inline nmdc_server/migrations/versions/988105f6581d_update_datasetDoi_to_awardDoi.py:10]: You can manually edit this line to be 'Revises: `last migration version#`

File: nmdc_server/migrations/versions/988105f6581d_update_datasetDoi_to_awardDoi.py
Comment (marySalvi) [inline nmdc_server/migrations/versions/988105f6581d_update_datasetDoi_to_awardDoi.py:26]: Again you can manually edit this line. No need to run a new migration.

Comment (JamesTessmer) [reply nmdc_server/migrations/versions/988105f6581d_update_datasetDoi_to_awardDoi.py:26]: Gotcha, thanks Mary. I'll edit those.

**Comment by JamesTessmer** [comment]
I've merged main back in and updated the migrator.

**Comment by mslarae13** [comment]
@JamesTessmer I have no comments on the PR. I trust @pkalita-lbl 's review from the technical perspective and i can test in -dev when this is ready.

File: web/src/views/SubmissionPortal/store/index.ts
Comment (JamesTessmer) [reply web/src/views/SubmissionPortal/store/index.ts]: Applied all the above changes

**Comment by JamesTessmer** [comment]
> @JamesTessmer I have no comments on the PR. I trust @pkalita-lbl 's review from the technical perspective and i can test in -dev when this is ready.

Cool, I will hit the merge button now.

**Pull request merged**
